index_configuration_tool/search_endpoint.py

```diff
@@ -0,0 +1,37 @@
+import sys
+from typing import Optional
+
+import requests
+
+# Constants
+SETTINGS_KEY = "settings"
+MAPPINGS_KEY = "mappings"
+__INDEX_KEY = "index"
+__ALL_INDICES_ENDPOINT = "*"
+__INTERNAL_SETTINGS_KEYS = ["creation_date", "uuid", "provided_name", "version", "store"]
+
+
+def fetch_all_indices(endpoint: str, optional_auth: Optional[tuple] = None) -> dict:
+    actual_endpoint = endpoint + __ALL_INDICES_ENDPOINT
+    resp = requests.get(actual_endpoint, auth=optional_auth)
+    # Remove internal settings
+    result = dict(resp.json())
+    for index in result:
+        for setting in __INTERNAL_SETTINGS_KEYS:
+            index_settings = result[index][SETTINGS_KEY]
+            if __INDEX_KEY in index_settings:
+                index_settings[__INDEX_KEY].pop(setting, None)
```

Comment:
the second param of pop just sets a default return value for the function (so if `setting` wasn't in the dictionary, it would return `None` instead of the value of the `setting` key). In this case, you're not doing anythign with the returned value (you're just treating pop as delete), so you don't need a default value at all.

Reply:
Without the default value, the `dict` would throw a `KeyError` if the key is not found ([docs](https://python-reference.readthedocs.io/en/latest/docs/dict/pop.html_)) 😞